### PR TITLE
[irods/irods#7905] Document MariaDB compatibility situation

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Addresses irods/irods#7905

### CODE REVIEW

File: docs/getting_started/installation.md
Comment (korydraughn) [inline docs/getting_started/installation.md:98]: `--transaction-isolation` -> `--transaction_isolation`

https://mariadb.com/docs/server/ref/mdb/system-variables/tx_isolation

File: docs/getting_started/installation.md
Comment (korydraughn) [inline docs/getting_started/installation.md:101]: These are the settings I used when running against the test suite. Notice the use of underscores just like with the transaction isolation level.
- `--character_set_server=utf8mb4`
- `--collation_server=utf8mb4_bin`

MariaDB didn't accept `utf8mb4_uca1400_as_cs` when I tried using that, but happily accepted the ones used above.

- https://mariadb.com/docs/server/ref/mdb/system-variables/character_set_server/
- https://mariadb.com/docs/server/ref/mdb/system-variables/collation_server/

Strangely enough, the docs I used were for MariaDB 10.6. They don't appear to have docs available for 11.4.

**Review by korydraughn — COMMENTED** [review]
Looks good overall.

File: docs/getting_started/installation.md
Comment (SwooshyCueb) [reply docs/getting_started/installation.md:98]: Both underscore and dash are valid https://mariadb.com/kb/en/server-system-variables/#tx_isolation

File: docs/getting_started/installation.md
Comment (SwooshyCueb) [reply docs/getting_started/installation.md:101]: Yes, MariaDB 10.6 doesn't have UCA-14.0.0 as it wasn't added until 10.10.1.
https://mariadb.com/kb/en/supported-character-sets-and-collations/

You can see in the testing environment that we are using `utf8mb4_bin` for 10.6 and `utf8mb4_uca1400_as_cs` for 10.11 and 11.4.

Also, as before, both dashes and underscores are valid:
https://mariadb.com/kb/en/server-system-variables/#character_set_server
https://mariadb.com/kb/en/server-system-variables/#collation_server

File: docs/getting_started/installation.md
Comment (korydraughn) [reply docs/getting_started/installation.md:98]: That's surprising. I don't remember the use of hyphens working for me. It's possible the values being passed to the options were invalid.

Will give it another try.

Comment (SwooshyCueb) [reply docs/getting_started/installation.md:98]: We are using a dash in the testing environment and it works just fine:
https://github.com/irods/irods_testing_environment/blob/main/projects/ubuntu-22.04/ubuntu-22.04-mariadb-10.6/docker-compose.yml#L11
https://github.com/irods/irods_testing_environment/blob/main/projects/ubuntu-22.04/ubuntu-22.04-mariadb-10.11/docker-compose.yml#L11
https://github.com/irods/irods_testing_environment/blob/main/projects/ubuntu-22.04/ubuntu-22.04-mariadb-11.4/docker-compose.yml#L11

File: docs/getting_started/installation.md
Comment (SwooshyCueb) [reply docs/getting_started/installation.md:101]: In the testing environment, we are using `utf8mb4_uca1400_as_cs` for 10.11 and 11.4, and `utf8mb4_bin` for 10.6 (also we're using dashes in the option names) and it works just fine:
https://github.com/irods/irods_testing_environment/blob/main/projects/ubuntu-22.04/ubuntu-22.04-mariadb-10.6/docker-compose.yml#L11
https://github.com/irods/irods_testing_environment/blob/main/projects/ubuntu-22.04/ubuntu-22.04-mariadb-10.11/docker-compose.yml#L11
https://github.com/irods/irods_testing_environment/blob/main/projects/ubuntu-22.04/ubuntu-22.04-mariadb-11.4/docker-compose.yml#L11

File: docs/getting_started/installation.md
Comment (korydraughn) [reply docs/getting_started/installation.md:98]: Dashes work for me too.

File: docs/getting_started/installation.md
Comment (korydraughn) [reply docs/getting_started/installation.md:101]: Let's stick with `utf8mb4_uca1400_as_cs` for now. If the tests fail during release testing of 4.3.3, we can discuss switching to `utf8mb4_bin`.

**Comment by korydraughn** [comment]
Let's keep this PR open until release testing for 4.3.3 + MariaDB is complete.

As soon as we confirm things are good for MariaDB, we can update the text (if necessary) and merge this.

**Comment by alanking** [comment]
I'm going to start running unit tests and core tests with the mariadb configurations in the testing environment, which are being documented here. Will report back with results as they appear.

**Comment by alanking** [comment]
I made a list of checkboxes in my previous comment, but just reporting that unit tests and core tests passed for all available versions of mariadb that we have in the testing environment for Ubuntu 22.04.

**Comment by alanking** [comment]
These are the projects in the testing environment as of the tip of main at time of writing (https://github.com/irods/irods_testing_environment/commit/13cb1a9d8b559d3498d465711fd833ad7ad26802).

10.6: https://github.com/irods/irods_testing_environment/blob/main/projects/ubuntu-22.04/ubuntu-22.04-mariadb-10.6/docker-compose.yml
```yaml
  catalog:
    image: mariadb:10.6
    environment:
      - MARIADB_ROOT_PASSWORD=testpassword
      - MARIADB_USER=irods
      - MARIADB_PASSWORD=testpassword
    # For parity with mysql 8.0, we use utf8mb4 charset and utf8mb4_bin collation (for case sensitivity), and disable the binary log
    command: "--transaction-isolation=READ-COMMITTED --character-set-server=utf8mb4 --collation-server=utf8mb4_bin --disable-log-bin"
```
Here are the relevant bits:
```
--transaction-isolation=READ-COMMITTED
```
This matches the configuration described on line 98 of the README in this PR: https://github.com/irods/irods_docs/pull/271/files#diff-753d13585ef80e9d9ccbe1d77bcd50095b7576cabf755f8bb7f49b5ade93d291R98
```
--character-set-server=utf8mb4
```
This matches the configuration described on line 101 of the README in this PR: https://github.com/irods/irods_docs/pull/271/files#diff-753d13585ef80e9d9ccbe1d77bcd50095b7576cabf755f8bb7f49b5ade93d291R101
```
--collation-server=utf8mb4_bin
```
This matches the configuration described on line 101 of the README in this PR: https://github.com/irods/irods_docs/pull/271/files#diff-753d13585ef80e9d9ccbe1d77bcd50095b7576cabf755f8bb7f49b5ade93d291R101

10.11: https://github.com/irods/irods_testing_environment/blob/main/projects/ubuntu-22.04/ubuntu-22.04-mariadb-10.11/docker-compose.yml
```yaml
  catalog:
    image: mariadb:10.11
    environment:
      - MARIADB_ROOT_PASSWORD=testpassword
      - MARIADB_USER=irods
      - MARIADB_PASSWORD=testpassword
    # For parity with mysql 8.0, we use utf8mb4 charset and utf8mb4_uca1400_as_cs collation (for case sensitivity), and disable the binary log
    command: "--transaction-isolation=READ-COMMITTED --character-set-server=utf8mb4 --collation-server=utf8mb4_uca1400_as_cs --disable-log-bin"
```
Here are the relevant bits:
```
--transaction-isolation=READ-COMMITTED
```
This matches the configuration described on line 98 of the README in this PR: https://github.com/irods/irods_docs/pull/271/files#diff-753d13585ef80e9d9ccbe1d77bcd50095b7576cabf755f8bb7f49b5ade93d291R98
```
--character-set-server=utf8mb4
```
This matches the configuration described on line 101 of the README in this PR: https://github.com/irods/irods_docs/pull/271/files#diff-753d13585ef80e9d9ccbe1d77bcd50095b7576cabf755f8bb7f49b5ade93d291R101
```
--collation-server=utf8mb4_uca1400_as_cs
```
This matches the configuration described on line 101 of the README in this PR: https://github.com/irods/irods_docs/pull/271/files#diff-753d13585ef80e9d9ccbe1d77bcd50095b7576cabf755f8bb7f49b5ade93d291R101

11.4: https://github.com/irods/irods_testing_environment/blob/main/projects/ubuntu-22.04/ubuntu-22.04-mariadb-11.4/docker-compose.yml
```yaml
  catalog:
    image: mariadb:11.4
    environment:
      - MARIADB_ROOT_PASSWORD=testpassword
      - MARIADB_USER=irods
      - MARIADB_PASSWORD=testpassword
    # For parity with mysql 8.0, we use utf8mb4 charset and utf8mb4_uca1400_as_cs collation (for case sensitivity), and disable the binary log
    command: "--transaction-isolation=READ-COMMITTED --character-set-server=utf8mb4 --collation-server=utf8mb4_uca1400_as_cs --disable-log-bin"
```
Here are the relevant bits:
```
--transaction-isolation=READ-COMMITTED
```
This matches the configuration described on line 98 of the README in this PR: https://github.com/irods/irods_docs/pull/271/files#diff-753d13585ef80e9d9ccbe1d77bcd50095b7576cabf755f8bb7f49b5ade93d291R98
```
--character-set-server=utf8mb4
```
This matches the configuration described on line 101 of the README in this PR: https://github.com/irods/irods_docs/pull/271/files#diff-753d13585ef80e9d9ccbe1d77bcd50095b7576cabf755f8bb7f49b5ade93d291R101
```
--collation-server=utf8mb4_uca1400_as_cs
```
This matches the configuration described on line 101 of the README in this PR: https://github.com/irods/irods_docs/pull/271/files#diff-753d13585ef80e9d9ccbe1d77bcd50095b7576cabf755f8bb7f49b5ade93d291R101

I see no inconsistencies between what is already in the testing environment (which is what I used to run the tests) and what is documented in this PR.

File: docs/getting_started/installation.md
Comment (korydraughn) [reply docs/getting_started/installation.md:101]: Recent testing by @alanking has shown that the configuration captured by this documentation aligns with the testing environment.

Marking as resolved.

**Review by korydraughn — APPROVED** [review]
I think this is ready for merging.

Let's wait for one more approval before pounding.

**Review by alanking — APPROVED** [review]
Agreed. # it.

**Comment by SwooshyCueb** [comment]
\#'d

**Comment by korydraughn** [comment]
Merging.

**Pull request merged**
